biogeochem/FatesSoilBGCFluxMod.F90

```diff
@@ -0,0 +1,1092 @@
+             cpatch => cpatch%younger
+          end do
+       end if
+       
+       
+       ! If nutrient competition is sent to the BGC model as PFTs
+       ! and not as individual cohorts, we need to unravel the input
+       ! boundary condition and send to cohort.  We do this downscaling
+       ! by finding each cohort's fraction of total fine-root for the group
+       
+       if(n_uptake_mode.eq.coupled_n_uptake .or. p_uptake_mode.eq.coupled_p_uptake)then
+
+          ! Note there are two scaling methods.  Either competition for
+          ! N and/or P was performed by cohorts acting individually
+          ! (cohort_ncomp_scaling) , or as PFTs (pft_ncomp_scaling)
+          ! If we opt for the latter, then we assume that the nutrient
+          ! uptake share of the cohort, matches the fraction of root
+          ! mass it contributes to the group (PFT).
+          
+          if(fates_ncomp_scaling.eq.pft_ncomp_scaling) then
+             
+             ! *Currently, all cohorts in a PFT have the same root
+             ! fraction, so all we have to to is find its total mass fraction.
+             
+             fnrt_c_pft(:) = 0._r8
+             cpatch => sites(s)%oldest_patch
+             do while (associated(cpatch))
+                ccohort => cpatch%tallest
+                do while (associated(ccohort))
+                   pft   = ccohort%pft
+                   fnrt_c_pft(pft) = fnrt_c_pft(pft) + &
+                        ccohort%prt%GetState(fnrt_organ, all_carbon_elements)*ccohort%n
+                   ccohort => ccohort%shorter
+                end do
+                cpatch => cpatch%younger
+             end do
+             
+          end if  ! end if(fates_ncomp_scaling.eq.pft_ncomp_scaling) then
+          
+          ! --------------------------------------------------------------------------------
+          ! Now that we have the arrays ready for downscaling (if needed)
+          ! loop through all cohorts and acquire nutrient
+          ! --------------------------------------------------------------------------------
+
+          if(n_uptake_mode.eq.coupled_n_uptake) then
+             icomp = 0
+             cpatch => sites(s)%oldest_patch
+             do while (associated(cpatch))
+                ccohort => cpatch%tallest
+                do while (associated(ccohort))
+                   pft   = ccohort%pft
+                   if(fates_ncomp_scaling.eq.cohort_ncomp_scaling) then
+                      icomp = icomp+1
+
+                      ! N Uptake:  Convert g/m2/day -> kg/plant/day
+                      ccohort%daily_n_uptake = ccohort%daily_n_uptake + &
+                           sum(bc_in(s)%plant_n_uptake_flux(icomp,:))*kg_per_g*AREA/ccohort%n
+
+                   else
+                      icomp = pft
+                      ! Total fine-root carbon of the cohort [kgC/ha]
+                      fnrt_c   = ccohort%prt%GetState(fnrt_organ, all_carbon_elements)*ccohort%n
```

Comment:
since we don't have isotopes in the allocation algorithm, there is no difference. But I think we should convert all instances of "all_carbon_elements" to carbon12_element until that day comes

Reply:
i propose we do that in another PR though, it will change a lot of lines